src/useD2.js

```diff
@@ -4,23 +4,31 @@ import { useConfig } from '@dhis2/app-runtime'
 
 let theD2 = null
 
-const configI18n = async (baseUrl, i18nRoot) => {
+const configI18n = async (baseUrl, i18nRoot, locale) => {
     config.baseUrl = baseUrl
 
-    const settings = await getUserSettings()
+    if (!locale) {
+        const settings = await getUserSettings()
 
-    if (settings.keyUiLocale && settings.keyUiLocale !== 'en') {
-        config.i18n.sources.add(
-            `${i18nRoot}/i18n_module_${settings.keyUiLocale}.properties`
-        )
+        locale = settings.keyUiLocale
```

Comment:
This is a mutation on one of the arguments. Would it be better to duplicate the `config.i18n.sources.add` line instead?

Reply:
I think this is OK, it could possibly be a new variable if you don't want to overwrite the argument, which is dissuaded by some eslint configurations - `const currentLocale = locale || (await getUserSettings()).keyUiLocale`

Reply:
Fixed.